src/DataTypes/Serializations/SerializationDecimalBase.cpp

```diff
@@ -34,23 +34,34 @@ void SerializationDecimalBase<T>::serializeBinaryBulk(const IColumn & column, Wr
 
     if (limit == 0 || offset + limit > size)
         limit = size - offset;
-
-    ostr.write(reinterpret_cast<const char *>(&x[offset]), sizeof(FieldType) * limit);
+    if constexpr (std::endian::native == std::endian::big && sizeof(T) >= 2)
+    {
+        for (size_t i = 0; i < limit; i++)
+        {
+            auto tmp(x[offset+i]);
+            char *start = reinterpret_cast<char*>(&tmp);
+            char *end = start + sizeof(FieldType);
+            std::reverse(start, end);
```

Comment:
You should be able to use `transformEndianness` instead of this from `src/Common/TransformEndianness.hpp`.

Reply:
Tried, but somehow the templates don't resolve for decimal types. so let's keep it as it is.